File: packages/ui-ios/Sources/ChorusUICatalog/CatalogSection.swift
import SwiftUI
import ChorusTokens

/// Labelled vertical group used by every catalog screen. Title sits at
/// the role's `Label.sm` token, content below at the stack-md gap.
struct CatalogSection<Content: View>: View {
    let title: String
    @ViewBuilder var content: () -> Content

    var body: some View {
        VStack(alignment: .leading, spacing: ChorusLayout.Stack.sm) {
            Text(title)
                .font(.system(size: ChorusTypography.Label.sm.size,
                              weight: ChorusTypography.Label.sm.fontWeight))
                .foregroundStyle(.secondary)
            content()
        }
    }
}

File: packages/ui-android/chorus-ui-catalog/src/main/kotlin/dev/blinddsai/chorus/ui/catalog/CatalogSection.kt
package dev.blinddsai.chorus.ui.catalog

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import dev.blinddsai.chorus.tokens.ChorusTheme
import dev.blinddsai.chorus.tokens.generated.ChorusLayout
import dev.blinddsai.chorus.tokens.generated.ChorusTypography

/** Labelled vertical group used by every catalog screen. */
@Composable
internal fun CatalogSection(title: String, content: @Composable () -> Unit) {
    Column(verticalArrangement = Arrangement.spacedBy(ChorusLayout.Stack.sm)) {
        Text(title, style = ChorusTypography.Label.sm, color = ChorusTheme.colors.onSurfaceVariant)
        content()
    }
}
